secure-api-gateway-ob-uk-common-obie-datamodel/src/main/java/uk/org/openbanking/datamodel/account/OBCurrencyExchange5.java

```diff
@@ -210,7 +212,8 @@ public boolean equals(Object o) {
         return Objects.equals(this.sourceCurrency, obCurrencyExchange5.sourceCurrency) &&
                 Objects.equals(this.targetCurrency, obCurrencyExchange5.targetCurrency) &&
                 Objects.equals(this.unitCurrency, obCurrencyExchange5.unitCurrency) &&
-                Objects.equals(this.exchangeRate, obCurrencyExchange5.exchangeRate) &&
+                // TODO: temporary fix for https://github.com/SecureApiGateway/SecureApiGateway/issues/981
+                isEqual(this.exchangeRate, obCurrencyExchange5.exchangeRate) &&
```

Comment:
Modification in `equals` method to verify the equality of BigDecimals.

`isEqual(BigDecimal compareTo, BigDecimal toCompare)` imported as static, used in all OB objects that have a BigDecimal field.